### PR TITLE
Add `#[required(<count>)]`

### PR DESCRIPTION
**Pull request by littledivy**
Equivalent to Deno's `webidl.requiredArguments` internal.

```rust
#[fast]
#[required(2)]
fn set_item(
  &self,
  state: &mut OpState,
  #[string] key: &str,
  #[string] value: &str,
) {
 // ...
}
```

### CODE REVIEW

File: ops/op2/dispatch_slow.rs [hunk 1]
```diff
@@ -227,6 +235,34 @@ pub(crate) fn with_fn_args(
...
+  required: u8,
+) -> TokenStream {
+  generator_state.needs_scope = true;
+  let arguments_lit = if required > 1 {
+    "arguments"
+  } else {
+    "argument"
+  };
+  gs_quote!(generator_state(fn_args, scope, self_ty) =>
+    (if #fn_args.length() < #required as i32 {
```
[the remark this answers is not in the document]
Comment (littledivy) [reply ops/op2/dispatch_slow.rs:249]: `#fn_args.length() < #required` => `1 < 2` so it won't pass

File: ops/op2/dispatch_slow.rs [hunk 2]
```diff
@@ -227,6 +235,34 @@ pub(crate) fn with_fn_args(
...
+      let msg = format!(
+        "Failed to execute '{}' on '{}': {} {} required, but only {} present",
```
Comment (crowlKats) [inline ops/op2/dispatch_slow.rs:251]: two notes on this:
- `Failed to execute '{}' on '{}':` we will need to extract this out when we add more of the webidl conversion so we dont duplicate this. no action needed now, just noting for future.
- this is only applicable for methods. could you handle normal functions as well? ie https://github.com/denoland/deno/blob/main/ext/canvas/01_image.js#L158-L166
- Same applies for constructors, as those say `Failed to construct '{}':`

Comment (littledivy) [reply ops/op2/dispatch_slow.rs:251]: - noted.
- ah didn't know that. this pr is only for object wraps but easy to have this on functions too.
- yup will fix that, ig it also needs to be 'Failed to construct {}.{}` for static methods

**Review by crowlKats — APPROVED** [review]
LGTM

**Pull request merged**